File: sqlitemagic-tests/app/src/main/java/com/siimkinks/sqlitemagic/transformer/ObjectTransformer.java
package com.siimkinks.sqlitemagic.transformer;

import android.support.annotation.NonNull;

import com.siimkinks.sqlitemagic.annotation.transformer.DbValueToObject;
import com.siimkinks.sqlitemagic.annotation.transformer.ObjectToDbValue;
import com.siimkinks.sqlitemagic.model.TransformableObject;

public class ObjectTransformer {

  @NonNull
  @ObjectToDbValue
  public static Integer objectToDbValue(TransformableObject javaObject) {
    return javaObject.value;
  }

  @NonNull
  @DbValueToObject
  public static TransformableObject dbValueToObject(Integer dbObject) {
    return new TransformableObject(dbObject);
  }
}

File: sqlitemagic-tests/app-kotlin/src/main/kotlin/com/siimkinks/sqlitemagic/transformer/ObjectTransformer.kt
package com.siimkinks.sqlitemagic.transformer

import com.siimkinks.sqlitemagic.annotation.transformer.DbValueToObject
import com.siimkinks.sqlitemagic.annotation.transformer.ObjectToDbValue
import com.siimkinks.sqlitemagic.model.TransformableObject

@ObjectToDbValue
fun objectToDbValue(javaObject: TransformableObject): Int {
  return javaObject.value
}

@DbValueToObject
fun dbValueToObject(dbObject: Int): TransformableObject {
  return TransformableObject(dbObject)
}